helper/winsvc/mock_windows_service.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package winsvc
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/shoenig/test/must"
+)
+
+func NewMockWindowsServiceManager(t *testing.T) *MockWindowsServiceManager {
+	return &MockWindowsServiceManager{t: t}
+}
+
+func NewMockWindowsService(t *testing.T) *MockWindowsService {
+	return &MockWindowsService{t: t}
+}
+
+type MockWindowsServiceManager struct {
+	isServiceRegistereds [][3]any
+	getServices          [][3]any
+	createServices       [][5]any
```

Comment:
these `[x]any`s become hard to reason about with stuff like

```go
	if call[4] == nil {
		return call[3].(WindowsService), nil
	}
```

this mock is fancy enough that I think 3 little structs for these things would make things more readable without appreciably adding more complexity (and it'd avoid potential panics from bad type assertions), so then that example chunk would read something like

```go
	if call.err == nil {
		return call.windowsSvc, nil
	}
```